test/Microsoft.AspNetCore.SignalR.Tests.Utils/TestClient.cs

```diff
@@ -47,11 +48,24 @@ public TestClient(bool synchronousCallbacks = false, IHubProtocol protocol = nul
 
             _cts = new CancellationTokenSource();
 
-            using (var memoryStream = new MemoryStream())
+        }
+
+        public async Task<Task> ConnectAsync(dynamic endPoint, bool sendNegotiateMessage = true)
```

Comment:
Can we make this `ConnectAsync<THub>(HubEndpoint<THub> endPoint, ...)` ? Though I have a vague feeling I may have tried this very refactoring in a spike and ended up swallowing my pride and using `dynamic` as well...

Reply:
Ugh, I remember now, `HubEndPointTestUtils` returns `dynamic`... Ok, let's leave fixing that for another day :)

Reply:
If there was a non-generic base class or interface for `HubEndpoint<T>` then we could pass that around instead of `dynamic`.